Bumping version of MessagePack breaks source generation inside VisualStudio

## Actual behavior

f25bc2d updated MessagePack version from `2.2.x` to `2.3.x` which [has some breaking changes](https://github.com/neuecc/MessagePack-CSharp/releases/tag/v2.3.75) regarding serialization of `Dictionary<>`.

Simple example fails to generate sources inside VisualStudio:
`CSC : warning CS8785: Generator 'SourceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'MessagePackSerializationException' with message 'Failed to deserialize System.Collections.Generic.Dictionary2[[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[Microsoft.Windows.SDK.Win32Docs.ApiDetails, Microsoft.Windows.SDK.Win32Docs, Version=0.1.4.39555, Culture=neutral, PublicKeyToken=7c40c7c34b1fcfe1]] value.`

Compiling project from command line with `dotnet build` works without problem - that's probably why all automatic tests also works.
I've recompiled CsWin32 with MessagePack version downgraded to `2.2.x` and it works again.

## Repro steps

1. `NativeMethods.txt` content:
```
CreateFile
```

2. Empty console project with `using Windows.Win32;`

### Context

- CsWin32 version: 0.1.539-beta
- Target Framework: net5.0
- VisualStudio: 2019 and 2022

--- a/.github/dependabot.yml
+++ b/.github/dependabot.yml
@@ -7,3 +7,5 @@ updates:
   directory: /
   schedule:
     interval: monthly
+  ignore:
+  - dependency-name: MessagePack # We have to use the MessagePack version used by win32metadata (https://github.com/microsoft/CsWin32/issues/371)

--- a/src/Microsoft.Windows.CsWin32/Microsoft.Windows.CsWin32.csproj
+++ b/src/Microsoft.Windows.CsWin32/Microsoft.Windows.CsWin32.csproj
@@ -42,7 +42,7 @@
   </ItemDefinitionGroup>
 
   <ItemGroup>
-    <PackageReference Include="MessagePack" Version="2.3.75" />
+    <PackageReference Include="MessagePack" Version="2.2.85" />
     <PackageReference Include="Microsoft.CodeAnalysis.BannedApiAnalyzers" Version="3.3.2" PrivateAssets="all" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.8.0" />
     <PackageReference Include="Microsoft.Windows.SDK.Win32Metadata" Version="$(MetadataVersion)" GeneratePathProperty="true" PrivateAssets="none" />

--- a/test/Microsoft.Windows.CsWin32.Tests/Microsoft.Windows.CsWin32.Tests.csproj
+++ b/test/Microsoft.Windows.CsWin32.Tests/Microsoft.Windows.CsWin32.Tests.csproj
@@ -25,6 +25,7 @@
 
   <ItemGroup>
     <PackageReference Include="coverlet.msbuild" Version="3.1.0" />
+    <PackageReference Include="MessagePack" Version="2.2.85" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.10.0" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.SourceGenerators.Testing.XUnit" Version="1.1.1-beta1.21413.1" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.10.0" />

